arch/arm/soc/ti_simplelink/cc32xx/Kconfig.soc

```diff
@@ -15,9 +15,7 @@ endchoice
 if SOC_CC3220SF
 
 config CC3220SF_DEBUG
-	bool "Prepend debug header, disabling flash verification"
-	depends on XIP
-	default y if XIP
-	default n if !XIP
+	bool "Prepend debug header, disabling flash verification" if XIP
+	default XIP
```

Comment:
CC3220SF_DEBUG needs to be set 'y' if XIP.  
Doesn't this change break that logic?

Reply:
@GAnthony 
Will work the same way as before:

 * The `...if XIP` on the prompt says that `CC3220SF_DEBUG` can be only be changed in `menuconfig` (or in a `.conf` file) if `XIP` is `y`. Otherwise, the symbol is "invisible" in the configuration interface.

 * The `default XIP` says that the default value of the symbol is the value of `XIP`. The default value is used if the symbol is invisible or hasn't been set by the user (e.g. by toggling it with `[Space]` in the menuconfig).

It's easy to miss that prompts and defaults are independent like that, because `depends on` will put the same condition (`... if ...`) on everything. Whenever you have `depends on`, the prompt and the default will get disabled at the same time, but it's possible to disable/enable them separately as well.

Reply:
Ah, makes sense.  Thanks for making the improvement.